fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+        if (delinquencyBucket == null) {
+            throw DelinquencyBucketNotFoundException.notFound(delinquencyBucketId);
+        }
+        Map<String, Object> changes = new HashMap<>();
+        delinquencyBucket = updateDelinquencyBucket(delinquencyBucket, data, changes);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).with(changes)
+                .build();
+    }
+
+    @Override
+    public CommandProcessingResult deleteDelinquencyBucket(Long delinquencyBucketId, JsonCommand command) {
+        final DelinquencyBucket delinquencyBucket = repositoryBucket.getReferenceById(delinquencyBucketId);
+        repositoryBucket.delete(delinquencyBucket);
+        return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(delinquencyBucket.getId()).build();
+    }
+
+    @Override
+    @CronTarget(jobName = JobName.LOAN_DELINQUENCY_CLASSIFICATION)
+    public void setLoanDelinquencyClassification() throws JobExecutionException {
+
+    }
+
+    private DelinquencyRange updateOrCreateDelinquencyRange(DelinquencyRangeData data, Map<String, Object> changes) {
+        Optional<DelinquencyRange> delinquencyRange = repositoryRange.findByClassification(data.getClassification());
+
+        if (delinquencyRange.isEmpty()) {
+            DelinquencyRange newDelinquencyRange = DelinquencyRange.instance(data.getClassification(), data.getMinimumAge(),
+                    data.getMaximumAge());
+            return repositoryRange.saveAndFlush(newDelinquencyRange);
+        } else {
+            return updateDelinquencyRange(delinquencyRange.get(), data, changes);
+        }
+    }
+
+    private DelinquencyRange updateDelinquencyRange(DelinquencyRange delinquencyRange, DelinquencyRangeData data,
+            Map<String, Object> changes) {
+        if (!data.getClassification().equalsIgnoreCase(delinquencyRange.getClassification())) {
+            delinquencyRange.setClassification(data.getClassification());
+            changes.put(DelinquencyApiConstants.CLASSIFICATION_PARAM_NAME, data.getClassification());
+        }
+        if (!data.getMinimumAge().equals(delinquencyRange.getMinimumAge())) {
+            delinquencyRange.setMinimumAge(data.getMinimumAge());
+            changes.put(DelinquencyApiConstants.MINIMUMAGE_PARAM_NAME, data.getMinimumAge());
+        }
+        if (!data.getMaximumAge().equals(delinquencyRange.getMaximumAge())) {
+            delinquencyRange.setMaximumAge(data.getMaximumAge());
+            changes.put(DelinquencyApiConstants.MAXIMUMAGE_PARAM_NAME, data.getMaximumAge());
+        }
+        if (!changes.isEmpty()) {
+            delinquencyRange = repositoryRange.saveAndFlush(delinquencyRange);
+        }
+        return delinquencyRange;
+    }
+
+    private DelinquencyBucket updateOrCreateDelinquencyBucket(DelinquencyBucketData data, Map<String, Object> changes) {
```

Comment:
Same here.

Reply:
The same as above, same recent implementation as BusinessDate